sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -201,6 +203,14 @@ class V2ExpressionBuilder(
         None
       }
     // TODO supports other expressions
+    case ApplyFunctionExpression(function, children) =>
+      val childrenExpressions = children.flatMap(generateExpression(_))
+      if (childrenExpressions.length == children.length) {
+        Some(new GeneralScalarExpression(function.name().toUpperCase(Locale.ROOT),
```

Comment:
UDF is kind of special and I think it's better to have a new v2 API for it, instead of reusing `GeneralScalarExpression`.

e.g., we can add a `ScalarUDFExpression`, which defines the function name, canonical name and inputs.